ui/load/ticket.go

```diff
@@ -0,0 +1,169 @@
+	Status     string
+	Fee        string
+	Amount     string
+	DateTime   string
+	MonthDay   string
+	DaysBehind string
+	WalletName string
+
+	timestamp int64
+}
+
+const (
+	StakingLive     = "LIVE"
+	StakingImmature = "IMMATURE"
+	StakingExpired  = "EXPIRED"
+	StakingRevoked  = "REVOKED"
+	StakingVoted    = "VOTED"
+)
+
+func (wl *WalletLoad) StakingOverviewAllWallets() *dcrlibwallet.StakingOverview {
+	overview := new(dcrlibwallet.StakingOverview)
+	for _, w := range wl.MultiWallet.AllWallets() {
+		ov, _ := w.StakingOverview()
+		overview.All += ov.All
+		overview.Expired += ov.Expired
+		overview.Immature += ov.Immature
+		overview.Live += ov.Live
+		overview.Revoked += ov.Revoked
+		overview.Voted += ov.Voted
+	}
+
+	wl.MultiWallet.GetLowestBlockTimestamp()
+	return overview
+}
+
+func calculateDaysBehind(lastHeaderTime int64) string {
+	diff := time.Since(time.Unix(lastHeaderTime, 0))
+	daysBehind := int(math.Round(diff.Hours() / 24))
+	if daysBehind < 1 {
+		return "<1 day"
+	} else if daysBehind == 1 {
+		return "1 day"
+	} else {
+		return fmt.Sprintf("%d days", daysBehind)
+	}
+}
+
+func filterToStatus(txFilter int32) string {
+	switch txFilter {
+	case dcrlibwallet.TxFilterImmature:
+		return StakingImmature
+	case dcrlibwallet.TxFilterLive:
+		return StakingLive
+	case dcrlibwallet.TxFilterExpired:
+		return StakingExpired
+	case dcrlibwallet.TxFilterRevoked:
+		return StakingRevoked
+	case dcrlibwallet.TxFilterVoted:
+		return StakingVoted
+	}
+
+	return ""
+}
+
+func transactionToTicket(tx dcrlibwallet.Transaction, status, walletName string) Ticket {
+	return Ticket{
+		Status:     status,
+		Amount:     dcrutil.Amount(tx.Amount).String(),
+		DateTime:   time.Unix(tx.Timestamp, 0).Format("Jan 2, 2006 03:04:05 PM"),
+		MonthDay:   time.Unix(tx.Timestamp, 0).Format("Jan 2"),
+		DaysBehind: calculateDaysBehind(tx.Timestamp),
+		Fee:        dcrutil.Amount(tx.Fee).String(),
+		WalletName: walletName,
+		timestamp:  tx.Timestamp,
+	}
+}
+
+func transactionsToTickets(txs []dcrlibwallet.Transaction, status, walletName string) []Ticket {
+	var tickets []Ticket
+	for _, tx := range txs {
+		tickets = append(tickets, transactionToTicket(tx, status, walletName))
+	}
+
+	return tickets
+}
+
+func (wl *WalletLoad) getAllTickets(walletID int, newestFirst bool) ([]Ticket, error) {
+	w := wl.MultiWallet.WalletWithID(walletID)
+	var tickets []Ticket
+
+	addTickets := func(txFilter int32, newestFirst bool) error {
+		txs, err := w.GetTransactionsRaw(0, 0, txFilter, newestFirst)
+		if err != nil {
+			return err
+		}
+
+		tickets = append(tickets, transactionsToTickets(txs, filterToStatus(txFilter), w.Name)...)
+		return nil
+	}
+
+	filters := []int32{
+		dcrlibwallet.TxFilterImmature,
+		dcrlibwallet.TxFilterLive,
+		dcrlibwallet.TxFilterVoted,
+		dcrlibwallet.TxFilterExpired,
+		dcrlibwallet.TxFilterRevoked,
+	}
+
+	for _, filter := range filters {
+		err := addTickets(filter, newestFirst)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Why not?
```go
txs, err := w.GetTransactionsRaw(0, 0, dcrlibwallet.TxFilterStaking, newestFirst)
if err != nil {
    return nil, err
}
```

Reply:
currently the ticket status such as immature and expired tickets can't be gotten directly from the transaction without calculation. Except for voted and revoked tickets which are set as transaction types.